tests/Unit/Elliptic/DiscontinuousGalerkin/Test_ImposeBoundaryConditions.cpp

```diff
@@ -0,0 +1,487 @@
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementIndex<Dim>;
+  using const_global_cache_tag_list = tmpl::list<NumericalFluxTag>;
+  using action_list = tmpl::list<
+      Elliptic::dg::Actions::ImposeHomogeneousDirichletBoundaryConditions<
+          Metavariables>,
+      dg::Actions::ReceiveDataForFluxes<Metavariables>>;
+
+  using flux_comm_types = dg::FluxCommunicationTypes<Metavariables>;
+  using simple_tags = db::AddSimpleTags<
+      TemporalId, Tags::Next<TemporalId>, Tags::Mesh<Dim>, Tags::Element<Dim>,
+      Tags::ElementMap<Dim>, bdry_vars_tag,
+      bdry_normal_dot_fluxes_tag<flux_comm_types>, bdry_other_data_tag,
+      exterior_bdry_normal_dot_fluxes_tag<flux_comm_types>,
+      exterior_bdry_other_data_tag, exterior_bdry_vars_tag,
+      mortar_data_tag<flux_comm_types>, mortar_next_temporal_ids_tag,
+      mortar_meshes_tag, mortar_sizes_tag>;
+
+  using compute_tags = db::AddComputeTags<
+      Tags::BoundaryDirectionsInterior<Dim>,
+      boundary_compute_tag<Tags::Direction<Dim>>,
+      boundary_compute_tag<Tags::InterfaceMesh<Dim>>,
+      boundary_compute_tag<Tags::UnnormalizedFaceNormal<Dim>>,
+      boundary_compute_tag<
+          Tags::EuclideanMagnitude<Tags::UnnormalizedFaceNormal<Dim>>>,
+      boundary_compute_tag<Tags::Normalized<Tags::UnnormalizedFaceNormal<Dim>>>,
+      Tags::BoundaryDirectionsExterior<Dim>,
+      exterior_boundary_compute_tag<Tags::Direction<Dim>>,
+      exterior_boundary_compute_tag<Tags::InterfaceMesh<Dim>>,
+      exterior_boundary_compute_tag<Tags::BoundaryCoordinates<Dim>>,
+      exterior_boundary_compute_tag<Tags::UnnormalizedFaceNormal<Dim>>,
+      exterior_boundary_compute_tag<
+          Tags::EuclideanMagnitude<Tags::UnnormalizedFaceNormal<Dim>>>,
+      exterior_boundary_compute_tag<
+          Tags::Normalized<Tags::UnnormalizedFaceNormal<Dim>>>>;
+
+  using initial_databox =
+      db::compute_databox_type<tmpl::append<simple_tags, compute_tags>>;
+};
+
+struct Metavariables {
+  using system = System;
+  using component_list = tmpl::list<ElementArray<Metavariables>>;
+  using temporal_id = TemporalId;
+  using const_global_cache_tag_list = tmpl::list<>;
+
+  using normal_dot_numerical_flux = NumericalFluxTag;
+};
+
+template <typename Component>
+using compute_items = typename Component::compute_tags;
+
+SPECTRE_TEST_CASE("Unit.Elliptic.DG.Actions.BoundaryConditions",
+                  "[Unit][Elliptic][Actions]") {
+  using flux_comm_types = typename ElementArray<Metavariables>::flux_comm_types;
+  using my_component = ElementArray<Metavariables>;
+  const Mesh<2> mesh{3, Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto};
+
+  //      xi      Block       +- xi
+  //      |     0   |   1     |
+  // eta -+ +-------+-+-+---+ eta
+  //        |       |X| |   |
+  //        |       +-+-+   |
+  //        |       | | |   |
+  //        +-------+-+-+---+
+  // We run the actions on the indicated element.
```

Comment:
"on the element indicated on the diagram." Now, I have seen this diagram in the code before, but I remember others and myself having trouble deciphering it, so I would suggest adding a bit more description on how to read the diagram.

Currently I don't remember how to read the diagram myself :stuck_out_tongue: but I will make a guess:
There are two elements - the element on the left (element 0) has its xi axis pointing upward and eta axis pointing to the left, while the element on the right (element 1) has its xi axis pointing to the right and eta axis pointing downwards.  I remember the first time I deciphered this diagram that it took me a while to even get that far.

Is the indicated element the element indicated by the `x`? I would use the wording "`self_id` is the ElementId (given by the Block ID and segment IDs) of the element indicated by the `x`. After that the rest becomes clear :smile:

Reply:
There are two blocks (not two elements), xi is pointing right and eta is pointing down. I made a new diagram which is simpler and hopefully clearer.